apps/prairielearn/assets/scripts/instructorAssessmentAccessClient.ts

```diff
@@ -0,0 +1,225 @@
+
+  const ptHost = (accessRulesTable as HTMLElement)?.dataset.ptHost ?? '';
+  const devMode = (accessRulesTable as HTMLElement)?.dataset.devMode === 'true';
+  const hasCourseInstancePermissionView =
+    (accessRulesTable as HTMLElement)?.dataset.hasCourseInstancePermissionView === 'true';
+  const timezone = (accessRulesTable as HTMLElement)?.dataset.timezone ?? 'UTC';
+
+  let editMode = false;
+
+  function refreshTable() {
+    morphdom(
+      accessRulesTable as Node,
+      AccessRulesTable({
+        accessRules: accessRulesData,
+        ptHost,
+        devMode,
+        hasCourseInstancePermissionView,
+        editMode,
+      }).toString(),
+    );
+  }
+
+  refreshTable();
+
+  const addRuleButton = document.getElementById('addRuleButton');
+
+  enableEditButton?.addEventListener('click', () => {
+    editMode = true;
+    enableEditButton.style.display = 'none';
+    editModeButtons?.style.removeProperty('display');
+    addRuleButton?.style.removeProperty('display');
+    refreshTable();
+  });
+
+  on('click', '#saveAndSyncButton', () => {
+    const form = document.getElementById('accessRulesForm') as HTMLFormElement;
+    const assessmentAccessRulesInput = form.querySelector(
+      'input[name="assessment_access_rules"]',
+    ) as HTMLInputElement;
+    if (!assessmentAccessRulesInput) return;
+    accessRulesData.forEach((accessRule: Record<string, any>) => {
+      accessRule.startDate = adjustedDate(accessRule.start_date).toISOString().slice(0, 19);
+      accessRule.endDate = adjustedDate(accessRule.end_date).toISOString().slice(0, 19);
+      accessRule.timeLimitMin = parseInt(accessRule.time_limit);
+      accessRule.uids = accessRule.uids.split(',').map((uid: string) => uid.trim());
+      accessRule.credit = parseInt(accessRule.credit);
+      if (accessRule.exam) {
+        accessRule.examUuid = accessRule.exam_uuid;
+      }
+      delete accessRule.formatted_start_date;
+      delete accessRule.formatted_end_date;
+      delete accessRule.start_date;
+      delete accessRule.end_date;
+      delete accessRule.time_limit;
+      delete accessRule.pt_course_name;
+      delete accessRule.pt_exam_name;
+      delete accessRule.pt_course_id;
+      delete accessRule.pt_exam_id;
+      delete accessRule.exam_uuid;
+      if (accessRule.active === 'True' || accessRule.active === 'true') {
+        accessRule.active = true;
+      } else {
+        accessRule.active = false;
+      }
+    });
+    assessmentAccessRulesInput.value = JSON.stringify(accessRulesData);
+    form.submit();
+  });
+
+  function adjustedDate(dateString: string | Date) {
+    const date = new Date(dateString);
+    const timezoneOffset = date.getTimezoneOffset() * 60000;
+    return new Date(date.getTime() - timezoneOffset);
+  }
+
+  let timeZoneName: string | undefined;
+  // To use Temporal polyfill, we need to check if Temporal is available in the environment
+  if (typeof Temporal !== 'undefined') {
```

Comment:
This is not true. `Temporal` here refers specifically to the imported value from `@js-temporal/polyfill`, not the `Temporal` global. Since we're explicitly using a polyfill, this check is not necessary.

Reply:
Removed this comment and check in [1dfef07](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/1dfef07527ed0296ff3cdfa7c58d85fce2304b97).